include/swift/AST/DiagnosticsSema.def

```diff
@@ -1054,6 +1054,8 @@ NOTE(single_confusable_character,none,
       (bool, StringRef, StringRef, StringRef, StringRef))
 ERROR(cannot_find_type_in_scope,none,
       "cannot find type %0 in scope", (DeclNameRef))
+ERROR(cannot_find_type_in_cast_expression,none,
+      "cast expression expects a type on its right-hand side (got: %0)", (DeclNameRef))
```

Comment:
Let's add DescriptiveDeclKind as the first argument here so the diagnostic can mention exactly what has been found instead of a type.

Reply:
Changed the message to match suggestion after discussion [here](https://github.com/apple/swift/issues/68825#issuecomment-1747729148)